File: Ackermann-function/Lua/ackermann-function.lua
function ack(M,N)
    if M == 0 then return N + 1 end
    if N == 0 then return ack(M-1,1) end
    return ack(M-1,ack(M, N-1))
end

for m=0,3 do
    for n=0,13 do
        print(ack(m,n))
    end
end

File: Ackermann-function/JavaScript/ackermann-function.js
function ack(m, n) {
 return m === 0 ? n + 1 : ack(m - 1, n === 0  ? 1 : ack(m, n - 1));
}

for(var m = 0; m < 4; m++){
    for(var n = 0; n < 14; n++){
        console.log(ack(m,n));
    }
}
